pkg/client/addon_deployment.go

```diff
@@ -43,9 +46,24 @@ func (h *V1Client) CreateAddonDeployment(uid string, body *models.V1SpectroClust
 		return nil
 	}
 
-	resolveNotification := true
+	resolveNotification := false // during initial creation we never need to resolve packs.
 	params := clusterC.NewV1SpectroClustersPatchProfilesParamsWithContext(h.Ctx).WithUID(uid).WithBody(body).WithResolveNotification(&resolveNotification)
-	_, err = client.V1SpectroClustersPatchProfiles(params)
+	err = patchWithRetry(h, err, client, params)
+	return err
+}
+
+func patchWithRetry(h *V1Client, err error, client clusterC.ClientService, params *clusterC.V1SpectroClustersPatchProfilesParams) error {
```

Comment:
passing in a nil error from outside `patchWithRetry` is not idiomatic go. it would be better to just return nil if the patch succeeds.

Reply:
made err variable local to the method.